includes/admin/class-wc-admin-dashboard-setup.php

```diff
@@ -0,0 +1,218 @@
+<?php
+/**
+ * Admin Dashboard - Setup
+ *
+ * @package     WooCommerce\Admin
+ * @version     2.1.0
+ */
+
+use Automattic\Jetpack\Constants;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+if ( ! class_exists( 'WC_Admin_Dashboard_Setup', false ) ) :
+
+	/**
+	 * WC_Admin_Dashboard_Setup Class.
+	 */
+	class WC_Admin_Dashboard_Setup {
+
+		/**
+		 * List of tasks.
+		 *
+		 * @var array
+		 */
+		private $tasks = array(
+			'store_details'        => array(
+				'completed'   => false,
+				'button_link' => 'admin.php?page=wc-admin&path=%2Fsetup-wizard',
+			),
+			'products'             => array(
+				'completed'   => false,
+				'button_link' => 'admin.php?page=wc-admin&task=products',
+			),
+			'woocommerce-payments' => array(
+				'completed'   => false,
+				'button_link' => 'admin.php?page=wc-admin&path=%2Fpayments%2Fconnect',
+			),
+			'payments'             => array(
+				'completed'   => false,
+				'button_link' => 'admin.php?page=wc-admin&task=payments',
+			),
+			'tax'                  => array(
+				'completed'   => false,
+				'button_link' => 'admin.php?page=wc-admin&task=tax',
+			),
+			'shipping'             => array(
+				'completed'   => false,
+				'button_link' => 'admin.php?page=wc-admin&task=shipping',
+			),
+			'appearance'           => array(
+				'completed'   => false,
+				'button_link' => 'admin.php?page=wc-admin&task=appearance',
+			),
+		);
+
+		/**
+		 * # of completed tasks.
+		 *
+		 * @var int
+		 */
+		private $completed_tasks_count = 0;
+
+		/**
+		 * WC_Admin_Dashboard_Setup constructor.
+		 */
+		public function __construct() {
+			if ( $this->should_display_widget() ) {
+				$this->populate_general_tasks();
+				$this->populate_payment_tasks();
+				$this->completed_tasks_count = $this->get_completed_tasks_count();
+				$this->init();
```

Comment:
`init()` looks like it could just be inlined here, rather than have a seperate function

Reply:
Makes sense. I will make an update 👍